### DISCUSSION TITLE
Pass "page file" and QS to shorturl?

### DISCUSSION THREAD

**Discussion by karl0ss** [YOURLS/YOURLS#3121] [Q&A]
Just found this great software, but am running into some troubles, and hope someone can shed some light for me...

I have setup a redirect 

- `http://mydom.com/ABC`

This will redirect to a site that I have to pass some QS to, but it is also progrmaticly adding the php file on to the url, so my ORG url should be something like 

- `http://otherdom.com/xyz.php?qs1=test`

My current redirect does 

- `http://mydom.com/ABC -> http://otherdom.com`

This does infact take me to the root as expected....but when I now try to pass this onto my shorturl 

- `http://mydom.com/ABC/xyz.php?qs1=test`

I am not redirected and brought to a 403 Forbidden on my own server, not the "hoped" for location of -

- `http://otherdom.com/xyz.php?qs1=test`

I'm assuming it can't handle the 
- /xyz.php

I have tried using 

- Append Query String
- Keep-query-string
- qs_forward
- Query String Keeper

but none of these seemed to make the change I needed...

Hope someone has an idea, or if im barking up the wrong tree and this will never work...

basically i want to be able to accept

`/xyz.php?qs1=test`

being passed onto the short url and it redirect to 

`http://otherdom.com/xyz.php?qs1=test`

Thanks

Karl

**Comment by dgw** [comment]
Those query-string plugins don't do what you want because you want to pass along more than just a query string.

I'd bet a plugin that does do this exists, or could be written… but honestly, the simplest solution would be to implement that kind of redirect at the HTTPd (Apache, Nginx, Caddy, whatever) layer. It's just a prefix, and what you lose in convenience of managing them from a YOURLS GUI you'd more than make up for in performance. Not to mention the development time saved.

**Reply by karl0ss** [reply]
@dgw  Thank you for getting back to me

Yeah I'm strating to think this isn't going to be as easy as I thought it would be, yourls seemed a good choice as i can programaticly update the urls and stuff....

real shame, I will carry on having a play

I guess if I can interogate 

`$_SERVER['REQUEST_URI']`

to see if there is a `/` after the "first whole word" (using regex) i could then chop after that and pass on in the REDIRECT_QUERY_STRING and pass only the first half to REDIRECT_URL would that create the correct query to pull out the correct redirection url, then it would appeant the "new qs" that contains say the page path?

**Reply by dgw** [reply]
Probably some voodoo with filters throughout the redirection chain, starting with `shunt_keyword_is_taken` and continuing all the way through the redirection process. Follow it into `yourls-go.php` and the filters in functions called from there.

I half want to map it out for you, but it's not really practical to do from my phone and it sounds like you'll learn a lot from poking through the code to follow all the steps involved in matching keywords and making the redirection happen. ☺

**Reply by karl0ss** [reply]
Thank you for all your help, 

I will continue to have a poke about..
